feat: OPTIC-1360: [BE] create Account settings endpoints that handle existing functionality in the django template

mcanu opened the pull request:
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [x] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)



#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [x] Backend (API)
- [ ] Frontend



### Describe the reason for change
Adding endpoint for getting the needed information about a user in an organization for the new Settings page.


### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [x] No
- [ ] Not sure (briefly explain the situation below)



### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [ ] integration
- [x] unit


- https://github.com/HumanSignal/label-studio-sdk/pull/364 linked by [Workflow Run](https://github.com/HumanSignal/label-studio/actions/runs/12145328900)

mcanu committed:
feat: OPTIC-1360: [BE] create Account settings endpoints that handle existing functionality in the django template

mcanu committed:
Added date_joined to whoami endpoint. Added created_at to organizations endpoint.

mcanu committed:
Fix permissions

mcanu committed:
Added tests

mcanu committed:
Added swagger schema

fern-api[bot] committed:
[submodules] Bump HumanSignal/label-studio-sdk version

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/12145328900

mcanu committed:
[submodules] Bump HumanSignal/label-studio-sdk version

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/12146196095

wesleylima commented on label_studio/organizations/serializers.py:
```diff
@@ -104,6 +98,23 @@ class Meta:
+
+    def get_annotations_count(self, member):
+        org = self.context.get('organization')
+        return member.user.annotations.filter(project__organization=org).count()
```
I am a little worried about the performance of this count. This will execute a new query for every user in the organization.
Is this similar to how the django template was handling this? If that is the case, then we can address performance optimization in a future ticket.

wesleylima replied:
Oh I just realized this is for the "detail" view so this would be the count for only one org member. This is totally fine!

mcanu replied:
In the [template](https://github.com/HumanSignal/label-studio/blob/7479e3e82cc78e97d42949f414813629ddde84b3/label_studio/users/templates/users/user_account.html#L214-L215) is using [these two methods](https://github.com/HumanSignal/label-studio/blob/d14d01e0409389ccd26254f9c54d4776acd5d67f/label_studio/users/models.py#L143-L149) in the user model. The only difference is that in the model is for the active organization, here is for the provided one.

wesleylima commented:
Left a small question but overall looks great! Good job @mcanu !

bmartel commented on label_studio/tests/webhooks/organizations.tavern.yml:
```diff
@@ -102,7 +102,7 @@ stages:
 - name: soft_delete_user_fails_second_deletion_attempt
   request:
-    url: "{django_live_url}/api/users/{user_pk}/soft-delete"
+    url: "{django_live_url}/api/organizations/{org_pk}/memberships/{user_pk}"
```
Nice catch!

robot-ci-heartex committed:
[submodules] Bump HumanSignal/label-studio-sdk version

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/12166670839

mcanu commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/12166805876)
> Successfully merged:  create mode 100644 web/libs/frontend-test/src/helpers/utils/fixLSParams.ts

mcanu committed:
Merge branch 'develop' into 'fb-optic-1360'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/12166805876
